### DISCUSSION TITLE
injecting telemetry messages with Cosmos

### DISCUSSION THREAD

**Discussion by mogar** [OpenSatKit/OpenSatKit#48] [Q&A]
I'm prototyping a system with OpenSatKit, and I'd like to inject some telemetry messages with cosmos for testing. Is there a way to do this easily?

The only way I've found so far is to turn the telemetry message into a command, which would then send a different packet header. That would require changes to my software, but would let me test my app functionality. Is there a better way of doing it?

**Comment by dmccomas** [comment]
Nothing currently exists on the COSMOS side that I'm aware of so a custom
script would have to be written.  COSMOS has a way to playback telemetry
into its own tools but an adapter would have to be written to route it back
to the FSW.

Are you interested in playing back historical telemetry or do you need a
telemetry packet source that generates test telemetry?   If it's the latter
then modifying a simple app like OSK's Instrument Sim (ISIM) app to create
test data would be an easy route.

- Dave
